pkg/kubecost/allocation.go

```diff
@@ -435,6 +489,19 @@ func (a *Allocation) add(that *Allocation) {
 	a.LoadBalancerCost += that.LoadBalancerCost
 	a.SharedCost += that.SharedCost
 	a.ExternalCost += that.ExternalCost
+
+	if addType == accumulation {
+		// Accumulation-type additions imply same resource, so the overall
+		// maximum is the correct resulting value.
+		a.CPUCoreUsageMax = math.Max(a.CPUCoreUsageMax, that.CPUCoreUsageMax)
+		a.RAMBytesUsageMax = math.Max(a.RAMBytesUsageMax, that.RAMBytesUsageMax)
+	} else if addType == aggregation {
+		// Aggregation-type additions imply different resources, same
+		// aggregation group, so the correct maximum is actually the sum
+		// of maxima.
+		a.CPUCoreUsageMax += that.CPUCoreUsageMax
+		a.RAMBytesUsageMax += that.RAMBytesUsageMax
+	}
```

Comment:
Totally understand the reason this was introduced, but I have a few concerns. (Not least that we used to do this, and it made working with the `Allocation` type more complicated having to know _how_ you wanted to add two of them together. Things are very nice when `Add` is smart and the type is resilient.)

1. We (probably) have internal information that allows us to make this decision. We have `window` to apply exactly this set of logic; i.e. if the windows do not overlap, then we are accumulating; otherwise, we are aggregating.
2. This proposed logic (or even following the logic above) may not work. Consider aggregating by controller, and we come across the `kubecost-cost-analyzer` pod. We find 10 pods, and we're aggregating, so we add their maxes. But they ran serially! This is also by-definition, not coincidence, because they share the same controller. So we'll be giving 10x recommendations for this controller. There are further issues we may run into on this front, which we should discuss, but if you start by simply trying to add the maxes for everything in a window, you might end up with funky results.